### DISCUSSION TITLE
OOT module don't see #include <img_mgmt/image.h>

### DISCUSSION THREAD

**Discussion by CreativeIndustryAg** [zephyrproject-rtos/zephyr#56652] [General]
Hello

I have to prepare own DFU protocol, so I implement it in project directory and it works. 
But when I move this functionality to OOT module, then compilation return error:

                fatal error: img_mgmt/image.h: No such file or directory
                   11 | #include <img_mgmt/image.h>_

The code (except moving it to OOT module) is the same and the problem is only with this file, rest are findable.  
It seems somehow CMake can't see it or I don't know. 

Could you help me?

**Comment by yashi** [comment]
That's because mcumgr sets include directories for the `app`, and `app` only.

https://github.com/zephyrproject-rtos/zephyr/blob/07c6af3b8c35c1e49186578ca61a25c76e2fb308/subsys/mgmt/mcumgr/CMakeLists.txt#L9

https://github.com/zephyrproject-rtos/zephyr/blob/07c6af3b8c35c1e49186578ca61a25c76e2fb308/cmake/modules/extensions.cmake#L590-L618

**Reply by CreativeIndustryAg** [reply]
@yashi You are the MEN :)

Many thanks. 
Do you think is this possible to use this macro inside OOT module?

**Reply by yashi** [reply]
Sure can.  But that doesn't allow your module to find mgmt include directory.

I haven't tested but if you are creating a library as an out-of-tree module, you need to do something like:

```cmake
# Let the app know where to find your include directory
# This is only for the include dirs.
zephyr_interface_library_named(yourlib)
target_include_directories(yourlib INTERFACE ${ZEPHYR_CURRENT_MODULE_DIR}/include)

# Now create the library itself.
zephyr_library()

# add your library source files to your library crated implictly by zephyr_library()
zephyr_library_sources(mylib.c)

# Your library need include directories from mgmt
zephyr_library_link_libraries(mgmt_mcumgr)
```

Search `zephyr_interface_library_named` under out of tree modules to see real world examples.

Hope it helps.

**Reply by CreativeIndustryAg** [reply]
@yashi Thanks a lot, I will test it after noon. 

Best wishes
